### DISCUSSION TITLE
Option to prevent the addition of virtual devices. HM-RCV-50/HmIP-RCV-50

### DISCUSSION THREAD

**Discussion by martinhoess** [SukramJ/aiohomematic#1947] [Ideas]
I'm currently looking for a way to prevent devices with virtual buttons/events from being added to Home Assistant. I came across the following PR/commit. https://github.com/SukramJ/hahomematic/pull/1070/files I'm not sure if it’s going in the right direction, but the commit is not present anymore in the current version anyway.

The main problem is that the integration creates 300 entities that I don't need. To make matters worse, the entities are created even if they are inactive/not visible in Homematic.

![image](https://github.com/user-attachments/assets/4a9ed920-32ce-4a13-873c-b0971e117422)
![image](https://github.com/user-attachments/assets/a9484ef1-b8be-4c54-8b5f-3ab557537124)

I understand what these buttons are intended for, but at the moment, I have absolutely no use for them. At the moment I just disable the entire device, but I'd prefer if the device and entities weren't created in HA in the first place.

**Comment by SukramJ** [comment]
Warum sollte es Sinn mach, dafür Zeit zu investieren?
Ob die Entitäten nun da sind, deaktiviert sind macht doch keinen Unterschied.

**Reply by martinhoess** [reply]
Weils für **mich** keinen Sinn macht da 300! Entitäten im System zu haben die ich nicht brauche. Wenn das 30 wären dann würde ich ja nix sagen. Und jedes Mal wenn man irgendwas (auf Dateiebene) sucht und/oder Filter falsch gesetzt hat dann stolpert man wieder über die gleichen Einträge. Für mich ist das ungefähr so Sinnvoll wie die HA fritz Integration die für jedes Netzwerkgerät 2 Entitäten anlegt. Das ist ein tolles Feature, für die Leute die es benötigen/verwenden für alle anderen müllt es halt nur das System unnötig zu 🤷

**Reply by SukramJ** [reply]
Der nächste braucht den Dutycyle oder die Betriebsspannung nicht, oder hätte statt der Entität lieber ein Attribut, und, und, und.
Also deaktiviere das Gerät, dann verschwinden die Entitäten aus den automatisch generierten Views, und gut ist.
Ich werde keine Zeit dafür investieren.
